ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -434,9 +434,15 @@ private void shutdown() {
         deleteManager.submitDeleteBlobOperation(blobId, new FutureResult<Void>(), null);
       }
       List<RequestInfo> requests = new ArrayList<RequestInfo>();
+      long lastTime = time.milliseconds();
       putManager.poll(requests);
+      routerMetrics.putManagerPollTimeMs.update(time.milliseconds() - lastTime);
+      lastTime = time.milliseconds();
       getManager.poll(requests);
+      routerMetrics.getManagerPollTimeMs.update(time.milliseconds() - lastTime);
+      lastTime = time.milliseconds();
       deleteManager.poll(requests);
+      routerMetrics.deleteManagerPollTimeMs.update(time.milliseconds() - lastTime);
       return requests;
```

Comment:
I think we usually follow the practice of tracking the time within the method rather than outside, and I feel that will also look cleaner.